packages/cardano-graphql-services/src/util/http.ts

```diff
@@ -1,9 +1,13 @@
 import { Application } from 'express';
 import http from 'http';
 
-export const appListenPromise = (app: Application, port: number, hostname: string): Promise<http.Server> =>
+export const listenPromise = (
+  serverLike: http.Server | Application,
+  port: number,
+  hostname?: string
+): Promise<http.Server> =>
   new Promise((resolve, reject) => {
-    const server: http.Server = app.listen(port, hostname, () => resolve(server));
+    const server = serverLike.listen(port, hostname!, () => resolve(server));
```

Comment:
This looks odd. Why is the `hostname` argument optional? Maybe use a default value instead, e.g. `'localhost'` or `''`

Reply:
It conforms with the overloaded type def:
```
listen(port?: number, hostname?: string, backlog?: number, listeningListener?: () => void): this;
listen(port?: number, hostname?: string, listeningListener?: () => void): this;
listen(port?: number, backlog?: number, listeningListener?: () => void): this;
listen(port?: number, listeningListener?: () => void): this;
listen(path: string, backlog?: number, listeningListener?: () => void): this;
listen(path: string, listeningListener?: () => void): this;
listen(options: ListenOptions, listeningListener?: () => void): this;
listen(handle: any, backlog?: number, listeningListener?: () => void): this;
listen(handle: any, listeningListener?: () => void): this;
```

... but you've highlighted an inconsistency, since port can also be optional. See my improved design here:
- e8b6f5bacb68e13e7e4f174932481a03e17745ce
- 47a407b46ecfe1eb5746a1f40c003719f332dc15

They're separate commits make it easily squashable

Reply:
Still not convinced it's correct, this overload: `listen(options: ListenOptions, listeningListener?: () => void): this;` doesn't accept `undefined` for `options`